index.js

```diff
@@ -245,7 +245,79 @@ const createClient = (profile, userAgent, opt = {}) => {
 		}
 	}
 
-	const refreshJourney = async (refreshToken, opt = {}) => {
+	const bestPrices = async (from, to, opt = {}) => {
+		from = profile.formatLocation(profile, from, 'from')
+		to = profile.formatLocation(profile, to, 'to')
+
+		opt = Object.assign({
+			via: null, // let journeys pass this station?
+			transfers: -1, // maximum nr of transfers
+			bike: false, // only bike-friendly journeys
+			tickets: false, // return tickets?
+			polylines: false, // return leg shapes?
+			subStops: false, // parse & expose sub-stops of stations?
+			entrances: false, // parse & expose entrances of stops/stations?
+			remarks: true, // parse & expose hints & warnings?
+			scheduledDays: false, // parse & expose dates each journey is valid on?
+		}, opt)
+		if (opt.via) opt.via = profile.formatLocation(profile, opt.via, 'opt.via')
+
+        let when = new Date();
+		if (opt.departure !== undefined && opt.departure !== null) {
+			when = new Date(opt.departure)
+			if (Number.isNaN(+when)) throw new TypeError('opt.departure is invalid')
+            const now = new Date();
+            const today = new Date(now.getFullYear(), now.getMonth(), now.getDate())
+            if (today > when) {
+                throw new TypeError('opt.departure date older than current date.')
+            }
+        }
+
+		const filters = [
+			profile.formatProductsFilter({profile}, opt.products || {})
+		]
+		if (
+			opt.accessibility &&
+			profile.filters &&
+			profile.filters.accessibility &&
+			profile.filters.accessibility[opt.accessibility]
+		) {
+			filters.push(profile.filters.accessibility[opt.accessibility])
+		}
+
+		const query = {
+			maxChg: opt.transfers,
+			depLocL: [from],
+			viaLocL: opt.via ? [{loc: opt.via}] : [],
+			arrLocL: [to],
+			jnyFltrL: filters,
+			getTariff: !!opt.tickets,
+
+			getPolyline: !!opt.polylines
+		}
+		query.outDate = profile.formatDate(profile, when)
+
+		const {res, common} = await profile.request({profile, opt}, userAgent, {
+			cfg: {polyEnc: 'GPA'},
+			meth: 'BestPriceSearch',
```

Comment:
I assume this call is only supported by the DB HAFAS endpoint?
If that is the case, we should make it `db`-profile-only (need to think about how).
If that is not the case, let's keep it in here, but also check *which* endpoints actually support it.

Reply:
I checked with cli.js that BestPriceSearch is only valid for the db profile.

> If that is the case, we should make it db-profile-only

Maybe it is sufficient to add a test that the current profile is the db profile.

It is ok that bestPrices is part of the optional api for hafasclient like trip or radar?